visualization/app/codeCharta/ui/fileChooser/fileChooser.component.ts

```diff
@@ -52,18 +52,38 @@ export class FileChooserController {
 					}
 
 					if (readFiles === element.files.length) {
-						this.setNewData()
+						this.setNewData(element.files)
 					}
 				}
 			}
 		})
 	}
 
-	setNewData() {
+	setNewData(sortingArray?) {
+		if (sortingArray !== undefined) {
+			this.sortFiles(sortingArray, this.files)
+		}
 		this.codeChartaService.loadFiles(this.files)
 		this.files = []
 	}
 
+	private findInFileList = (fileName: string, fileList: FileList): number => {
+		for (let index = 0; index < fileList.length; index++) {
+			if (fileList.item(index).name === fileName) {
+				return index
+			}
+		}
+		return 0
+	}
+
+	// this will sort the array based on the order the files have been imported, needed for e2e fileChooser test
+	private sortFiles(sortingArray, arrayToSort) {
+		const sorter = (a, b) => {
+			return this.findInFileList(a.fileName, sortingArray) - this.findInFileList(b.fileName, sortingArray)
+		}
+		return arrayToSort.sort(sorter)
+	}
```

Comment:
I'm not sure we should include private test-helperfunctions inside the component itself, as a developer I would assume that a private function is relevant for the components functionality. Could we move this to the test file or what do you think?

Reply:
I'm not sure about moving this to the test file, as we might want to have sorting on the real filechooser too.

Reply:
@BridgeAR what do you think?